pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AbstractNode.java

```diff
@@ -36,8 +37,8 @@
     protected int endColumn;
     private DataFlowNode dataFlowNode;
     private Object userData;
-    protected GenericToken firstToken;
-    protected GenericToken lastToken;
+    private GenericToken firstToken;
+    private GenericToken lastToken;
```

Comment:
why do we change the visibility of these? That's a breaking API change

Reply:
Just because these fields weren't being used in the project, but I haven't considered that perhaps someone with a custom rule is using any of these fields. I'll revert this.